### ISSUE TITLE
Quetsions about pre-processing and post-processing

### ISSUE THREAD

**Issue by l1346792580123** [humansensinglab/fabric-diffusion#1]
Thanks for opening source such great work! I have some questions about pre-processing and post-processing.

1. How to capture the regions of materials and prints? Is the example data extracted manually or obtained through some method? Does the proposed approach have any requirements on materials and prints regions?
2. Once the normalized texture is generated, how to map the rexture to a 3D mesh？How to determine the location of materials and prints if I want to reproduce the rendering results of the original image？

**Comment by HarryWang355** [comment]
Thank you for your interest in our work! To answer your questions:
1. The regional captures for materials and prints are manually selected. However, implementing an automated detection system for this process should be feasible. Each regional capture should focus on a single element of interest (e.g., texture or print) and be of a reasonable size.
2. In our paper, we used external software (Clo3D) to apply textures to garment assets. Alternatively, this can be done programmatically by mapping the tiled texture onto the mesh’s UV space. The texture and print locations need to be manually adjusted.
Let me know if you need further clarification!

**Comment by l1346792580123** [comment]
Thanks for your reply! Do you have any advice or clue to build an automated detection system? Should I use some corner detector or edge detector to find the regions of materials and prints?  I use blender to unwrap the garment mesh and map the reconstructed texture onto the UV space, while the result is not good since the uv space is not contiguous. How can I map the generated texture continuously onto arbitrary mesh? Moreover, do you have any plan to release the fine-tuned BRDF materials estimation model?

**Comment by HarryWang355** [comment]
You are welcome. We haven't extensively explored implementing such a detection system, but I can share a few thoughts. For prints, one should be able to finetune a standard object detection model using print data (people have built logo detection systems like this [one](https://clarifai.com/clarifai/main/models/logo-detection-v2)). For textures, if the reference image/object has a uniform texture pattern, generating regional captures should be as simple as generating random bounding boxes within the object mask; if multiple textures exist, then maybe we could try training a segmentation model to identify each texture region. I don't think edge detectors would work as some textures contain lots of noisy edges.
We recommend applying the generated textures onto garment assets with sewing patterns; unwrapping a raw garment mesh will certainly produce a fragmented UV map unsuitable for our purpose. To work with an arbitrary mesh, a clean and continuous UV map is necessary. Finally, yes, we do plan to release the finetuned BRDF estimation model.

**Comment by l1346792580123** [comment]
Thanks for your suggestions and looking forward to the BRDF model！
